pretixprint/app/src/main/java/eu/pretix/pretixprint/connections/System.kt

```diff
@@ -0,0 +1,109 @@
+package eu.pretix.pretixprint.connections
+
+import android.content.Context
+import android.os.Bundle
+import android.os.CancellationSignal
+import android.os.ParcelFileDescriptor
+import android.print.*
+import android.print.PrintAttributes.*
+import com.lowagie.text.pdf.PdfReader
+import eu.pretix.pretixprint.PrintException
+import eu.pretix.pretixprint.R
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileOutputStream
+
+class SystemConnection : ConnectionType {
+    override val identifier = "system"
+    override val nameResource =  R.string.connection_type_system
+    override val inputType = ConnectionType.Input.PDF
+
+    override fun allowedForUsecase(type: String): Boolean {
+        return type != "receipt"
+    }
+
+    override fun print(
+        tmpfile: File,
+        numPages: Int,
+        context: Context,
+        useCase: String,
+        settings: Map<String, String>?
+    ) {
+        val printManager = context.getSystemService(Context.PRINT_SERVICE) as PrintManager
+        val jobName = tmpfile.nameWithoutExtension
+
+        val `in` = FileInputStream(tmpfile)
+        val pdf = PdfReader(`in`)
+        var mediaSize : MediaSize = MediaSize.UNKNOWN_PORTRAIT
+        val rot = pdf.getPageRotation(1)
+        if (rot == 90 || rot == 180) {
```

Comment:
Good point. I've just created a landscape ticket layout and there the rotation was not set - because the media width is simply landscape and not a rotated portrait document :) fixed in 89ea387d